// ngx-audio-player/esm5/lib/service/audio-player-service/audio-player.service.js
import { __decorate, __metadata } from "tslib";
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
var AudioPlayerService = /** @class */ (function () {
    function AudioPlayerService() {
        this.playlist = [];
        this.indexSong = 0;
        this.currentTrack = new BehaviorSubject(this.playlist[this.indexSong]);
        this.currentTime = 0;
        this.duration = 0;
    }
    AudioPlayerService.prototype.init = function () {
        this.updateCurrentSong();
    };
    AudioPlayerService.prototype.nextSong = function () {
        if ((this.indexSong + 1) >= this.playlist.length) {
            this.indexSong = 0;
        }
        else {
            this.indexSong++;
        }
        this.updateCurrentSong();
    };
    AudioPlayerService.prototype.previousSong = function () {
        if ((this.indexSong - 1) < 0) {
            this.indexSong = (this.playlist.length - 1);
        }
        else {
            this.indexSong--;
        }
        this.updateCurrentSong();
    };
    AudioPlayerService.prototype.resetPlaylist = function () {
        this.indexSong = 0;
        this.updateCurrentSong();
    };
    AudioPlayerService.prototype.selectATrack = function (index) {
        this.indexSong = index - 1;
        this.updateCurrentSong();
    };
    AudioPlayerService.prototype.updateCurrentSong = function () {
        var current = this.playlist[this.indexSong];
        var previous = ((this.indexSong - 1) >= 0) ? this.playlist[this.indexSong - 1] : this.playlist[this.playlist.length - 1];
        var next = ((this.indexSong + 1) >= this.playlist.length) ? this.playlist[0] : this.playlist[this.indexSong + 1];
        this.currentTrack.next([
            previous,
            current,
            next
        ]);
    };
    AudioPlayerService.prototype.getSubjectCurrentTrack = function () {
        return this.currentTrack;
    };
    AudioPlayerService.prototype.getPlaylist = function () {
        return this.playlist;
    };
    AudioPlayerService.prototype.setPlaylist = function (playlist) {
        this.playlist = playlist;
    };
    AudioPlayerService.prototype.getIndexSong = function () {
        return this.indexSong;
    };
    AudioPlayerService = __decorate([
        Injectable(),
        __metadata("design:paramtypes", [])
    ], AudioPlayerService);
    return AudioPlayerService;
}());
export { AudioPlayerService };
